data-model/src/main/java/io/micronaut/data/model/runtime/EntityOperation.java

```diff
@@ -44,4 +45,10 @@
      */
     @Nullable
     StoredQuery<E, ?> getStoredQuery();
+
+    /**
+     * @return The invocation context
+     */
+    @NonNull
+    InvocationContext<?, ?> getInvocationContext();
```

Comment:
This might need to be nullable.

Reply:
Updated. I thought it would always have value since we read it from the method (MethodInvocationContext). But in these cases for this fix, it won't be null.